code/modules/antagonists/heretic/items/eldritch_painting.dm

```diff
@@ -0,0 +1,423 @@
+	timeout = 1 MINUTES
+
+// The special examine interaction for this painting
+/obj/structure/sign/painting/eldritch/desire/examine(mob/living/carbon/user)
+	if(IS_HERETIC(user))
+		// If they already have the negative moodlet return
+		if("heretic_eldritch_hunger" in user.mob_mood.mood_events)
+			return
+		// A list made of the organs and bodyparts the heretic possess
+		var/list/random_bodypart_or_organ = list(
+			/obj/item/organ/internal/brain,
+			/obj/item/organ/internal/lungs,
+			/obj/item/organ/internal/eyes,
+			/obj/item/organ/internal/ears,
+			/obj/item/organ/internal/heart,
+			/obj/item/organ/internal/liver,
+			/obj/item/organ/internal/stomach,
+			/obj/item/organ/internal/appendix,
+			/obj/item/bodypart/arm/left,
+			/obj/item/bodypart/arm/right,
+			/obj/item/bodypart/leg/left,
+			/obj/item/bodypart/leg/right
+		)
+		var/organ_or_bodypart_to_spawn = pick(random_bodypart_or_organ)
+		new organ_or_bodypart_to_spawn(drop_location())
+		to_chat(user, span_notice("A piece of flesh crawls out of the painting and flops onto the floor."))
+		// Adds a negative mood event to our heretic
+		user.add_mood_event("heretic_eldritch_hunger", /datum/mood_event/eldritch_painting/desire_heretic)
+
+	// Do they have the mood event added on examine, if so return
+	if ("respite_eldritch_hunger" in user.mob_mood.mood_events)
+		to_chat(user, span_notice("You are still full from your last viewing!"))
+		return
+
+	if (user.has_trauma_type(/datum/brain_trauma/severe/flesh_desire))
+		// Gives them some nutrition
+		user.adjust_nutrition(50)
+		to_chat(user, warning("You feel a searing pain in your stomach!"))
+		user.adjustOrganLoss(ORGAN_SLOT_STOMACH, 5)
+		to_chat(user, span_notice("You feel less hungry, but more empty somehow?"))
+		user.add_mood_event("respite_eldritch_hunger", /datum/mood_event/eldritch_painting/desire_examine)
+	return ..()
+
+// Specific proximity monitor for The First Desire or /obj/item/wallframe/painting/eldritch/desire
+/datum/proximity_monitor/advanced/eldritch_painting/desire
+	applied_trauma = /datum/brain_trauma/severe/flesh_desire
+	text_to_display = "What an artwork, just looking at it makes me hunger...."
+
+/*
+ * A brain trauma that this eldritch paintings apply
+ * This one is for "The First Desire" or /obj/structure/sign/painting/eldritch/desire
+ */
+/datum/brain_trauma/severe/flesh_desire
+	name = "The Desire for Flesh"
+	desc = "Patient seems to only be able to eat organs or raw flesh for nutrients, also seems to become hungrier at a faster rate"
+	scan_desc = "H_(82882)G3E:__))9R"
+	gain_text = span_warning("I feel a hunger, only organs and flesh will feed it...")
+	lose_text = span_notice("Your stomach no longer craves flesh, and your tongue feels duller.")
+	random_gain = FALSE
+	/// How much faster we loose hunger
+	var/hunger_rate = 15
+
+/datum/brain_trauma/severe/flesh_desire/on_gain()
+	// Allows them to eat faster, mainly for flavor
+	ADD_TRAIT(owner, TRAIT_VORACIOUS, "The Desire for Flesh")
```

Comment:
I kind of baited, you, the string key here is correct but we use defines for string keys. 
```dm
#define DESIRE_TRAIT "The Desire for Flesh"
...
ADD_TRAIT(owner, TRAIT_VORACIOUS, DESIRE_TRAIT )
```
However you can circumvent this by just using an easy to access key, such as name or type
```dm
ADD_TRAIT(owner, TRAIT_VORACIOUS, name)
```
or
```dm
ADD_TRAIT(owner, TRAIT_VORACIOUS, type)
```

In this case I recommend using name since it's the same thing

Reply:
As discussed on the discord I used ref(src)